src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/AggregationManager.cs

```diff
@@ -0,0 +1,314 @@
+        private Action<Instrument> _instrumentPublished;
+        private Action _initialInstrumentEnumerationComplete;
+        private Action<Exception> _collectionError;
+
+        public AggregationManager(
+            Action<Instrument, LabeledAggregationStatistics> collectMeasurement,
+            Action<DateTime> beginCollection,
+            Action<DateTime> endCollection,
+            Action<Instrument> beginInstrumentMeasurements,
+            Action<Instrument> endInstrumentMeasurements,
+            Action<Instrument> instrumentPublished,
+            Action initialInstrumentEnumerationComplete,
+            Action<Exception> collectionError)
+        {
+            _collectMeasurement = collectMeasurement;
+            _beginCollection = beginCollection;
+            _endCollection = endCollection;
+            _beginInstrumentMeasurements = beginInstrumentMeasurements;
+            _endInstrumentMeasurements = endInstrumentMeasurements;
+            _instrumentPublished = instrumentPublished;
+            _initialInstrumentEnumerationComplete = initialInstrumentEnumerationComplete;
+            _collectionError = collectionError;
+
+            _listener = new MeterListener()
+            {
+                InstrumentPublished = (instrument, listener) =>
+                {
+                    _instrumentPublished(instrument);
+                    InstrumentState? state = GetInstrumentState(instrument);
+                    if (state != null)
+                    {
+                        _beginInstrumentMeasurements(instrument);
+                        listener.EnableMeasurementEvents(instrument, state);
+                    }
+                },
+                MeasurementsCompleted = (instrument, cookie) =>
+                {
+                    _endInstrumentMeasurements(instrument);
+                    RemoveInstrumentState(instrument, (InstrumentState)cookie!);
+                }
+            };
+            _listener.SetMeasurementEventCallback<double>((i, m, l, c) => ((InstrumentState)c!).Update((double)m, l));
+            _listener.SetMeasurementEventCallback<float>((i, m, l, c) => ((InstrumentState)c!).Update((double)m, l));
+            _listener.SetMeasurementEventCallback<long>((i, m, l, c) => ((InstrumentState)c!).Update((double)m, l));
+            _listener.SetMeasurementEventCallback<int>((i, m, l, c) => ((InstrumentState)c!).Update((double)m, l));
+            _listener.SetMeasurementEventCallback<short>((i, m, l, c) => ((InstrumentState)c!).Update((double)m, l));
+            _listener.SetMeasurementEventCallback<byte>((i, m, l, c) => ((InstrumentState)c!).Update((double)m, l));
+            _listener.SetMeasurementEventCallback<decimal>((i, m, l, c) => ((InstrumentState)c!).Update((double)m, l));
+        }
+
+        public AggregationManager Include(string meterName)
+        {
+            Include(i => i.Meter.Name == meterName);
+            return this;
```

Comment:
Why do these Include methods return `this` if none of the call sites pay attention to the return value?

Reply:
Originally there was some code that was making calls in a fluent style. That code is gone now and 'return this' is a remnant that I forgot about. We can make it void.